src/Arcus.Messaging.Pumps.ServiceBus/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,175 @@
+            where TMessagePump : class, IHostedService
+        {
+            Guard.NotNull(services, nameof(services));
+
+            AddServiceBusMessagePump<TMessagePump>(services, subscriptionName: subscriptionName, getConnectionStringFromConfigurationFunc: getConnectionStringFromConfigurationFunc, configureMessagePump: configureMessagePump);
+
+            return services;
+        }
+
+        /// <summary>
+        ///     Adds a message handler to consume messages from Azure Service Bus Topic
+        /// </summary>
+        /// <param name="topicName">Name of the topic to work with</param>
+        /// <param name="subscriptionName">Name of the subscription to process</param>
+        /// <param name="services">Collection of services to use in the application</param>
+        /// <param name="getConnectionStringFromSecretFunc">Function to look up the connection string from the secret store</param>
+        /// <param name="configureMessagePump">Capability to configure how the message pump should behave</param>
+        /// <returns>Collection of services to use in the application</returns>
+        public static IServiceCollection AddServiceBusTopicMessagePump<TMessagePump>(this IServiceCollection services, string topicName, string subscriptionName, Func<ISecretProvider, Task<string>> getConnectionStringFromSecretFunc, Action<AzureServiceBusMessagePumpOptions> configureMessagePump = null)
+            where TMessagePump : class, IHostedService
+        {
+            Guard.NotNull(services, nameof(services));
+
+            AddServiceBusMessagePump<TMessagePump>(services, topicName, subscriptionName, getConnectionStringFromSecretFunc: getConnectionStringFromSecretFunc, configureMessagePump: configureMessagePump);
+
+            return services;
+        }
+
+        /// <summary>
+        ///     Adds a message handler to consume messages from Azure Service Bus Topic
+        /// </summary>
+        /// <param name="topicName">Name of the topic to work with</param>
+        /// <param name="subscriptionName">Name of the subscription to process</param>
+        /// <param name="services">Collection of services to use in the application</param>
+        /// <param name="getConnectionStringFromConfigurationFunc">Function to look up the connection string from the configuration</param>
+        /// <param name="configureMessagePump">Capability to configure how the message pump should behave</param>
+        /// <returns>Collection of services to use in the application</returns>
+        public static IServiceCollection AddServiceBusTopicMessagePump<TMessagePump>(this IServiceCollection services, string topicName, string subscriptionName, Func<IConfiguration, string> getConnectionStringFromConfigurationFunc, Action<AzureServiceBusMessagePumpOptions> configureMessagePump = null)
```

Comment:
The Arcus code guidelines talk about prevent using the optional parameters in public API, but I understand why it was used here; and it would only make the code here more duplicated/fragile.

Reply:
Is your concern here about `configureMessagePump`? I've tried to provide operations for given scenarios where `configureMessagePump` is always optional for when they're using a given scenario and want to tweak it.

We could provide an overload but will then have `nx2` amount of them, that might be too much maybe?

Reply:
Yeah, had the same though. Is good like it is for me.